src/motion/magic/Magic.tsx

```diff
@@ -337,18 +362,29 @@ export class Magic extends React.Component<FunctionalProps> {
         rotate.set(target)
     }
 
-    updateBorderRadius(p: number, borderRadius: MotionValue<string>) {
+    updateBorderRadius(
+        p: number,
+        borderRadius: MotionValue<string>,
+        index: number
+    ) {
+        const indexMap = [
```

Comment:
As `updateBorderRadius` runs four times per frame per component (actually currently potentially eight times as it might run again to account for parent distortion) I try and avoid creating data structures and functions in any "per frame" code. This is why the `boxShadow` updater is the way it is (in `createUpdateBoxShadow`) - its creating all the expensive stuff up front.

Also I don't think the array approach leads to any space savings so you could probably do without the array entirely, you could just make it more explicit and save the lookups with `updateBorderRadius(p, borderTopLeftRadius, "borderTopLeftRadius")`

Reply:
Good to know, great 👍